src/components/client/core/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,78 @@
+import { Portal } from "@ark-ui/react";
+import { FiX as CloseIcon } from "react-icons/fi";
+
+import { recipe as drawer } from "./Drawer.recipe";
+import { Button, Icon } from "components/client";
+import {
+  Drawer as PrimitiveDrawer,
+  DrawerBackdrop,
+  DrawerCloseTrigger,
+  DrawerContainer,
+  DrawerContent,
+  DrawerDescription,
+  DrawerTitle,
+  DrawerTrigger,
+} from "components/primitives";
+
+import type { DrawerProps } from "components/primitives";
+import type { ReactNode } from "react";
+
+export interface Props extends DrawerProps {
+  trigger: ReactNode;
+  // TODO: remove hard-coded values
+  placement?: "left" | "right";
```

Comment:
I think it's okay to keep this narrow type here and remove this TODO. I suppose you could extract it to an explicit type or even an enum if you'd like, but I think it's fine!

Reply:
Ok sounds good! Yeah I think I will remove for now, if we decide later that explicit types or enums are better, we can go that route too!